### PR TITLE
Support for text review commands on object name in Visual Studio Code

### LINKED ISSUE #16246: NVDA does not support text review commands in controls in Visual Studio Code.
I've prepared a PR to try and address this issue.
### Steps to reproduce:
Most controls in Visual Studio Code do not support text review commands.
1. Open the directory containing the file in VS Code.
2. Press Ctrl+Shift+E to focus the file tree.
3. Try using the Numpad1/Numpad3 text review commands to review file names character by character in the file tree.

### Actual behavior:
NVDA only reports the character '0xfffc'.

### Expected behavior:
NVDA should be able to report filenames character by character.

#### NVDA installed/portable/running from source:
installed, portable, running from source:
#### NVDA version:
All Version
#### Windows version:
Windows 10 22H2 (AMD64) build 19045.4046
#### Name and version of other software in use when reproducing the issue:
Visual Studio Code 1.86.2 
#### Other information about your system:
None
#### Does the issue still occur after restarting your computer?
Yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
All versions are the same
#### If NVDA add-ons are disabled, is your problem still occurring?
Yes
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
Yes

### PR DESCRIPTION
**Pull request by cary-rowen**
### Link to issue number:
Fixed #16246
### Summary of the issue:

Most controls in Visual Studio Code do not support text review commands.

The most typical example is when the user wants to review the control name character by character, such as the file name in the file tree and the setting items on the settings page. Based on current experience, this is not easy to achieve.

### Description of user facing changes
Users will be able to review control name in Visual Studio Code by NVDA’s text review commands.

### Description of development approach
In the application module for Visual Studio Code, a modification has been made within the  `event_NVDAObject_init`  method to enhance the handling of objects.
Specifically, for objects whose role is not  `EDITABLETEXT` , the  `TextInfo`  property is now explicitly set to  `NVDAObjectTextInfo`.
### Testing strategy:
Manual testing
### Known issues with pull request:
unknown
### Code Review Checklist:

- [X] Documentation:
  - Change log entry
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [X] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [X] UX of all users considered:
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English
- [ ] API is compatible with existing add-ons.
- [ ] Security precautions taken.

## Summary by CodeRabbit

- **New Features**
  - Introduced support for text review commands for an object's name in Visual Studio Code to enhance accessibility.
- **Bug Fixes**
  - Resolved issues related to NVDA interactions with Windows 11.

### CODE REVIEW

**Comment by CyrilleB79** [comment]
Is the issue being fixed here only applicable to VSCode?
Or could this fix be included in more general NVDA objects?

File: source/appModules/code.py [hunk 1]
```diff
@@ -1,15 +1,16 @@
...
 
 """ App module for Visual Studio Code.
 """
 
 import appModuleHandler
+import controlTypes
 from NVDAObjects.IAccessible.chromium import Document
 from NVDAObjects import NVDAObject
-
+from NVDAObjects import NVDAObjectTextInfo
```
Comment (seanbudd) [inline source/appModules/code.py:13]: this should be merged with the previous line, i.e.

```suggestion
from NVDAObjects import NVDAObject, NVDAObjectTextInfo
```

File: source/appModules/code.py [hunk 2]
```diff
@@ -24,3 +25,7 @@ class AppModule(appModuleHandler.AppModule):
 	def chooseNVDAObjectOverlayClasses(self, obj, clsList):
 		if Document in clsList and obj.IA2Attributes.get("tag") == "#document":
 			clsList.insert(0, VSCodeDocument)
+
+	def event_NVDAObject_init(self, obj):
```
Comment (seanbudd) [inline source/appModules/code.py:29]:
```suggestion
	def event_NVDAObject_init(self, obj: NVDAObject):
```

File: user_docs/en/changes.t2t [hunk 3]
```diff
@@ -49,6 +49,7 @@ What's New in NVDA
   - ``alt+upArrow`` is now mapped to ``dot2+dot3+dot7+space``
   - ``alt+downArrow`` is now mapped to ``dot5+dot6+dot7+space``
   -
+- Support for text review commands on object name in Visual Studio Code. (#16248, @Cary-Rowen)
```
Comment (seanbudd) [inline user_docs/en/changes.t2t:52]:
```suggestion
- Support added for text review commands for an object's name in Visual Studio Code. (#16248, @Cary-Rowen)
```

**Comment by CyrilleB79** [comment]
The issue being fixed here seems the same as the one in #14088.
In case the cause is the same, could this fix be generalized to more object types or duplicated for Chrome?

**Comment by cary-rowen** [comment]
Hi @CyrilleB79 
> The issue being fixed here seems the same as the one in #14088 .
> In case the cause is the same, could this fix be generalized to more object types or duplicated for Chrome?

I haven't investigated #14088 yet, could it be a duplicate of #15159?

**Comment by CyrilleB79** [comment]
> I haven't investigated #14088 yet, could it be a duplicate of #15159?

@cary-rowen, #14088 is about the menus in Chrome application itself, whereas #15159 is about web content displayed by the Chrome browser.
I cannot tell you if these two issues are due to the same root cause, but there is nothing obvious that would let it think.

Could you please have a deeper look at #14088 to see if it may be related to the current issue?

**Comment by cary-rowen** [comment]
#14088 can indeed be fixed using the same idea as this PR.

But I'd like to hear some suggestions on how to implement a more general fix.But I'd like to hear some best practices on how to implement a more general fix.

**Comment by CyrilleB79** [comment]
> #14088 can indeed be fixed using the same idea as this PR.
> 
> But I'd like to hear some best practices on how to implement a more general fix.

@seanbudd can you help here?

**Comment by seanbudd** [comment]
@coderabbitai can you suggest a more general fix using the idea from this PR to also fix #14088

Otherwise, I think this PR can be accepted as-is

**Comment by seanbudd** [comment]
@coderabbitai full review

**Comment by seanbudd** [comment]
@cary-rowen any thoughts on the general approach suggestions?

**Comment by cary-rowen** [comment]
I haven't tried it yet, based on the robot's suggestions:

>  Modify the Global Event Handler : Instead of modifying  event_NVDAObject_init  specifically in the application module for Visual Studio Code, this method can be adjusted in a more central location that handles all similar events across different applications.

Sounds reasonable, but I'm not sure if it's possible, I would really prefer to apply a more general approach to fix #14088

**Review by seanbudd — APPROVED** [review]

**Pull request merged**